third_party/proton/test/test_profile.py

```diff
@@ -57,6 +61,55 @@ def foo(x, y):
         assert data[0]["children"][1]["frame"]["name"] == "test2"
 
 
+def test_cudagraph():
+    if is_hip():
+        pytest.skip("HIP backend does not support profiling CUDA graphs")
+
+    stream = torch.cuda.Stream()
+    torch.cuda.set_stream(stream)
+
+    @triton.jit
+    def foo(x, y, z):
+        tl.store(z, tl.load(y) + tl.load(x))
+
+    def fn():
+        a = torch.ones((2, 2), device="cuda")
+        b = torch.ones((2, 2), device="cuda")
+        c = a + b
+        foo[(1, )](a, b, c)
+
+    with tempfile.NamedTemporaryFile(delete=True, suffix=".hatchet") as f:
+        proton.start(f.name.split(".")[0], context="shadow")
+
+        # warmup
+        # four kernels
+        fn()
+
+        # no kernels
+        g = torch.cuda.CUDAGraph()
+        with torch.cuda.graph(g):
+            for _ in range(10):
+                fn()
+
+        proton.enter_scope("test")
+        g.replay()
+        g.reset()
+        torch.cuda.synchronize()
+        proton.exit_scope()
+        proton.finalize()
+        data = json.load(f)
+        assert len(data[0]["children"]) == 5
+        # find the test frame
+        test_frame = None
+        for child in data[0]["children"]:
+            if child["frame"]["name"] == "test":
+                test_frame = child
+                break
+        assert test_frame is not None
+        assert len(test_frame["children"]) >= 4
+        assert test_frame["children"][0]["metrics"]["Time (ns)"] > 0
```

Comment:
I might also add the flops metric here (in addition to time) just as a check that the calc is actually being done (i.e. just "flops" > 0).